lib/sequence.js

```diff
@@ -291,21 +329,62 @@ module.exports = function SequenceFactory(connection) {
     const retriable = (cb) => {
       const id = this.getId();
       const referenceValue = this._getCounterReferenceField(doc);
+      const incAmount = this._options.inc_amount;
       this._counterModel.findOneAndUpdate(
         { id, reference_value: referenceValue },
-        { $inc: { seq: 1 } },
-        { new: true, upsert: true },
+        { $inc: { seq: incAmount } },
+        { new: true, upsert: false },
         (err, counter) => {
           if (err) return cb(err);
           return cb(null, counter.seq);
         },
-
       );
     };
 
     async.retry(0, retriable, callback);
   };
 
+  /**
+ * Utility function to create a record in counter before incrementing
+ *
+ * @method     _createCounter
+ * @param      {object}    doc       A mongoose model which need to receive the
+ *                                   increment
+ * @param      {Function}  callback  Called with the sequence counter
+ */
+  Sequence.prototype._createCounter = function (doc, callback) {
+    const id = this.getId();
+    const referenceValue = this._getCounterReferenceField(doc);
+    const startSeq = this._options.start_seq;
+    const counterModel = this._counterModel;
+
+    counterModel.findOneAndUpdate(
+      {
+        id,
+        reference_value: referenceValue,
+      },
+      {},
+      {
+        upsert: true,
+        new: true,
+        setDefaultsOnInsert: true,
+        rawResult: true,
+      },
+      (err, counter) => {
+        // mongodb issues parallel upsert with primary key
+        // #Ref: https://jira.mongodb.org/browse/SERVER-14322
+        // #Ref: https://docs.mongodb.com/manual/reference/method/db.collection.update/#use-unique-indexes
+        if (err && err.code !== 11000) {
```

Comment:
This check was needed as the test 'can create multiple document in parallel when the sequence is on _id' was failing.
I believe this is because of the mongodb issue that I have linked.